src/lua/LuaInput.cpp

```diff
@@ -321,6 +362,178 @@ static int l_input_set_joystick_enabled(lua_State *l)
 	return 0;
 }
 
+/*
+ * Function: ActionOnPress
+ *
+ * Activate the action corresponding to the given 'LuaInputAction' object
+ *
+ * Example:
+ *
+ * > action = Input.GetActionBinding("BindToggleSpeedLimiter")
+ * > if ui.isMouseClicked(0) then
+ * >     Input.ActionOnPress(action)
+ * > elseif ui.isMouseReleased(0) then
+ * >     Input.ActionOnRelease(action)
+ * > end
+ *
+ * Parameters:
+ *
+ *   action: a LuaInputAction object, obtained via Input.GetActionBinding
+ *
+ * Returns:
+ *
+ *   nothing
+ *
+ */
+static int l_input_action_on_press(lua_State *l)
+{
+	auto action = LuaObject<LuaInputAction>::CheckFromLua(1)->getAction();
+	action->binding.m_queuedEvents |= 1;
+	action->binding.m_active = true;
+	return 0;
+}
+
+/*
+ * Function: ActionOnRelease
+ *
+ * Deactivate the action corresponding to the given 'LuaInputAction' object
+ *
+ * Example:
+ *
+ * > action = Input.GetActionBinding("BindToggleSpeedLimiter")
+ * > if ui.isMouseClicked(0) then
+ * >     Input.ActionOnPress(action)
+ * > elseif ui.isMouseReleased(0) then
+ * >     Input.ActionOnRelease(action)
+ * > end
+ *
+ * Parameters:
+ *
+ *   action: a LuaInputAction object, obtained via Input.GetActionBinding
+ *
+ * Returns:
+ *
+ *   nothing
+ *
+ */
+static int l_input_action_on_release(lua_State *l)
+{
+	auto action = LuaObject<LuaInputAction>::CheckFromLua(1)->getAction();
+	action->binding.m_queuedEvents |= 2;
+	action->binding.m_active = false;
+	return 0;
+}
+
+/*
+ * Function: IsActionActive
+ *
+ * Check if the corresponding action is currently active
+ *
+ * Example:
+ *
+ * > action = Input.GetActionBinding("BindToggleSpeedLimiter")
+ * > active = Input.IsActionActive(action)
+ *
+ * Parameters:
+ *
+ *   action: a LuaInputAction object, obtained via Input.GetActionBinding
+ *
+ * Returns:
+ *
+ *   boolean
+ *
+ */
+static int l_input_is_action_active(lua_State *l)
+{
+	auto action = LuaObject<LuaInputAction>::CheckFromLua(1)->getAction();
+	// the action itself becomes active after dispatching, so we will check the
+	// corresponding bindings
+	bool active = action->binding.IsActive() || action->binding2.IsActive();
+	LuaPush(l, active);
+	return 1;
+}
+
+/*
+ * Function: IsActionActivated
```

Comment:
Instead of `IsActionActivated` vs `IsActionActive`, I'd recommend `IsActionJustActive`, mimicking the ImGui naming convention.

Reply:
Done! however, this has been moved to class member-function `IsActive` and `IsJustActive` due to the previous item.